src/client/dfuse/dfuse_bulk.h

```diff
@@ -0,0 +1,45 @@
+/**
```

Comment:
Eventually yes, but when at a point where we are able to port the code that currently uses it to DAOS arrays.  I'll do another check to see that all the code still here is necessary for compilation.

Reply:
> > I'm not sure what you mean by "we need to keep it for now until it's been replaced".
> > IMO the cart dependency should be removed and with that everything related to cart contexts, requests, RPC ops & in/out hdlr structs, bulk handling, crt_progress/wait calls, etc.
> 
> I don't think it would be possible to remove much more whilst still allowing the code to compile. MS3 deals with bootstrapping this code under DAOS and will replace a lot of the cart calls immediately.
> 

I guess i don't understand why. at least for things like the bulk handles and packing/unpacking structs, there should be no effect on compilation since by removing those, I would think you will also remove where they are used too.

> > I think this would reduce/simplify the patch considerably.
> 
> Agreed, it would, but it would also mean that a lot of the logic/work flow that we want to keep would be lost.

The way i see it now is that we are landing a lot of "dead" code into the repo that needs to be replaced later. I don't see how removing that would prevent compilation or effect the logic flow. instead of the calls to cart scheduling, progress, bulk processing, and all that, you would just have no-ops. I don't imagine those would work anyway now and that should not effect compilation the way i see it.
That being said, i don't want to impede your progress on MS3, so I'm not going to get hung up on that. I'll leave it to the gatekeeper to make a decision if this would be OK to land code that needs to be eventually replaced/removed.